src/App.tsx

```diff
@@ -1,26 +1,224 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import {
+  USER_ID,
+  deleteTodo,
+  getTodos,
+  postTodo,
+  setCompletedTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Errors } from './types/Error';
+import { Filter } from './types/Filter';
+import { filterTodo } from './utils/helpers';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<Errors | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deleteTodoId, setDeleteTodoId] = useState<number | null>(null);
+  const [title, setTitle] = useState('');
+  const filteredTodo = filterTodo(todos, filter);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const getData = async () => {
+      setIsLoading(true);
```

Comment:
Not necessarily a mistake here, but still, to follow the logic of try-catch-finally, this setter should be placed into `try` block